scripts/zkai_d128_recursive_pcd_route_selector_gate.py

```
@@ -0,0 +1,1015 @@
+        layer="source_evidence",
+    )
+    expect_equal(source.get("selected_slice_ids"), list(EXPECTED_SELECTED_SLICE_IDS), "selected slice ids", layer="source_evidence")
+    expect_equal(source.get("selected_checked_rows"), EXPECTED_SELECTED_ROWS, "selected rows", layer="source_evidence")
+    expect_equal(source.get("full_block_checked_rows"), EXPECTED_FULL_BLOCK_ROWS, "full block rows", layer="source_evidence")
+    require_commitment(source.get("two_slice_target_commitment"), "two_slice_target_commitment", layer="source_evidence")
+    require_commitment(source.get("two_slice_accumulator_commitment"), "two_slice_accumulator_commitment", layer="source_evidence")
+    require_commitment(source.get("two_slice_verifier_handle_commitment"), "two_slice_verifier_handle_commitment", layer="source_evidence")
+    return source
+
+
+def validate_local_repo_probe(value: Any) -> dict[str, Any]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)